libafl/src/events/simple.rs

```diff
@@ -382,11 +393,24 @@ where
 {
     fn process(
         &mut self,
-        fuzzer: &mut Z,
+        _fuzzer: &mut Z,
         state: &mut Self::State,
-        executor: &mut E,
+        _executor: &mut E,
     ) -> Result<usize, Error> {
-        self.simple_event_mgr.process(fuzzer, state, executor)
+        let count = self.simple_event_mgr.events.len();
+        // we need to override fn process and handle stop events appropriately
+        // since we are a restarting manager
+        while let Some(event) = self.simple_event_mgr.events.pop() {
```

Comment:
`pop` will turn around the order, I don't think it's worth it. Given that we will never have more than a hand full of events, just iterate twice

Reply:
This is now outdated since we will call ``on_shutdown`` on all managers before exiting anyways so we do not need to pre-process events if we use a non-restarting inner manager